pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -2554,13 +2554,18 @@ public void readEntryFailed(ManagedLedgerException exception, Object ctx) {
                     }, null);
             return future;
         } else {
-            Long ledgerId = ((ManagedCursorContainer) ledger.getCursors()).getSlowestReaderPosition().getLedgerId();
+            PositionImpl slowestPosition = ((ManagedCursorContainer) ledger.getCursors()).getSlowestReaderPosition();
+            Long ledgerId = slowestPosition.getLedgerId();
+            if (((ManagedLedgerImpl) ledger).getLedgersInfo().lastKey().equals(ledgerId)) {
+                return CompletableFuture.completedFuture(false);
+            }
             try {
                 org.apache.bookkeeper.mledger.proto.MLDataFormats.ManagedLedgerInfo.LedgerInfo
                         ledgerInfo = ledger.getLedgerInfo(ledgerId).get();
                 if (ledgerInfo != null && ledgerInfo.hasTimestamp() && ledgerInfo.getTimestamp() > 0
                         && ((ManagedLedgerImpl) ledger).getClock().millis() - ledgerInfo.getTimestamp()
-                        > backlogQuotaLimitInSecond * 1000) {
+                        > backlogQuotaLimitInSecond * 1000 && slowestPosition.compareTo(
+                                new PositionImpl(ledgerInfo.getLedgerId(), ledgerInfo.getEntries() - 1)) < 0) {
```

Comment:
Looks like this is an incorrect behavior here. If we have three ledgers, 0,1,2 and the `slowestPosition` reached the end of ledger 0, but ledger 1 also over the backlog quota limitation, we should return true here.

The correct behavior should be if the slowest position reaches the end of a ledger, we should calculate the backlog from the next ledger of the ledger of slowest position

Reply:
fixed.